Update Jest setup to use TypeScript

thibmeu opened the pull request:
Jest setup is the only file in JavaScript on the repository. This commit updates it to be a TypeScript file.

armfazh commented on test/jest.setup-file.ts:
```diff
@@ -3,8 +3,9 @@
 // Mocking crypto with NodeJS WebCrypto module only for tests.
-import { webcrypto } from 'node:crypto'
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { webcrypto } = require('node:crypto');
```
Using ESM, prefers the use of `import` instead of `require`

thibmeu replied:
updated. `jest.config.json` was pointing directly at the typescript file instead of the transpile version.

thibmeu committed:
Update Jest setup to use TypeScript

Jest setup is the only file in JavaScript on the repository. This commit
updates it to be a TypeScript file.
